### DISCUSSION TITLE
SOLVED: Very puzzling behaviour with a route in PWA

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#13015] [General - Components / Directives / etc] [answered]
I am building a Quasar PWA which I'm running into problems with in production which I can't make sense of. Let's say my app is at 'myapp.com'. If I go to 'myapp.com/register' ('/register' is one of the configured routes), I get an nginx 404 error, Not Found. But if I first go to 'myapp.com', load that, and then go to 'myapp.com/register' (in the browser address bar, not by clicking on a link), the page loads fine. Why would this be? And how can I get it so that I can go straight to 'myapp.com/register' without this intermediary step. Is this some PWA thing?

LATER: Duh! Turned out I'd screwed up the nginx config file and removed the necessary 'try_files' redirect. Still not sure why it worked quite as it did, but problem solved for now.

**Comment by rstoenescu** [comment, marked as answer]
Your nginx is not configured for vue router history mode.

Add this to your nginx config:

location / {
        try_files $uri $uri/ /index.html;
    }

**Reply by rstoenescu** [reply]
It's in docs too, at SPA > Deploying -- the concept is the same so we didn't duplicated the info.

**Reply by jonmor51** [reply]
Thanks, yes, I realised this just before you posted your response. I did originally have it configured correctly, which is why I didn't suspect nginx config, but somehow messed up the config with some recent changes.
